pkg/health/health.go

```diff
@@ -151,17 +171,38 @@ func consulWatchToResult(w kp.WatchResult) Result {
 
 // Returns the poorest status of all checks in the given list, plus the check
 // ID of one of those checks.
-func FindWorst(results []Result) (string, HealthState) {
-	worst := findWorstResult(results)
-	return worst.ID, worst.Status
+func FindWorst(results []Result) (string, HealthState, *HealthEmpty) {
+	if len(results) == 0 {
+		return "", Critical, &HealthEmpty{
+			Err: fmt.Errorf("no results were passed to FindWorst"),
+		}
+	}
+	worst, HEErr := findWorstResult(results)
+	if HEErr != nil {
+		return "", Critical, HEErr
+	}
+	return worst.ID, worst.Status, HEErr
 }
 
-func FindBest(results []Result) (string, HealthState) {
-	best := findBestResult(results)
-	return best.ID, best.Status
+func FindBest(results []Result) (string, HealthState, *HealthEmpty) {
+	if len(results) == 0 {
+		return "", Critical, &HealthEmpty{
+			Err: fmt.Errorf("no results were passed to FindBest"),
+		}
+	}
+	best, HEErr := findBestResult(results)
+	if HEErr != nil {
+		return "", Critical, HEErr
+	}
+	return best.ID, best.Status, HEErr
 }
 
-func findWorstResult(results []Result) Result {
+func findWorstResult(results []Result) (Result, *HealthEmpty) {
+	if len(results) == 0 {
+		return Result{Status: Critical}, &HealthEmpty{
+			Err: fmt.Errorf("no results were passed to findWorstResult"),
```

Comment:
Also, technically, you should make `HealthEmpty` conform to the `error` interface.